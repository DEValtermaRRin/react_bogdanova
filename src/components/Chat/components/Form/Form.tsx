import React, { useState, memo } from 'react';
import { Button } from './components/Button/Button';
import { ButtonDel } from './components/ButtonDel/ButtonDel';
import { Message } from './components/Message/Message';
import './Form.scss';

interface FormProps {
  addMessages: (val: string, user: string) => void;
  onSubmit?: () => void;
  userName: string;
  delMessages: () => void;
}

export const Form = memo<FormProps>(
  ({ delMessages, userName, addMessages }) => {
    const name = 'Send message';
    const [value, setValue] = useState('');

    const handleClickSubmit = (e: React.FormEvent<HTMLFormElement>) => {
      e.preventDefault();
      addMessages(value, userName);
      setValue('');
    };
    return (
      <form data-testid="form" onSubmit={handleClickSubmit} className="form">
        <ButtonDel delMessages={delMessages} />
        <Message value={value} setValue={setValue} />
        <Button disabled={!userName} name={name} />
      </form>
    );
  },
);
