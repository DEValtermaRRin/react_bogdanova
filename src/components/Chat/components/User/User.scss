.user {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 35px;
  height: 35px;
  background-color: #284779;
  overflow: hidden;
  border-radius: 15px;

  &:hover{
    animation: user-dragout 0.5s ease-in-out 0.1s 1 normal forwards;
  }

  &:hover &__input {
    animation: input-dragout 0.5s ease-in-out 0.1s 1 normal forwards;
  }
  
  &__icon {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 35px;
    height: 35px;
    background: url(./assets/user.svg) 100%;
    background-color: #284779;
    border-radius: 50%;
    overflow: hidden;
    &:hover {
      cursor: pointer;
    }
  }
  &__input {
    height: 30px;
    border-radius: 15px;
    border: none;
    background: #252424;
    font-size: 18px;
    font-weight: 600;
    letter-spacing: 0.1rem;
    color: #366dc5;
    opacity: 0;
    width: 0;
    margin-right: 0px;
        &::placeholder {
          padding-left: 5px;
      font-size: 16px;
      color: #6e6969;
      letter-spacing: 0.1rem;
    }
  }
}

@keyframes user-dragout {
  0% {
    width: 35px;
  }
  100% {
    width: 250px;
  }
}

@keyframes input-dragout {
  0% {
    padding: 0;
    opacity: 0;
    width: 0;
    margin-right: 0px;
  }
  100% {
    padding: 3px;
    opacity: 1;
    width: 200px;
    margin-right: 2px;
  }
}
