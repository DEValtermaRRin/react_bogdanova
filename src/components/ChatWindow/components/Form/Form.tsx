import React, { useState, memo } from 'react';
import { useDispatch } from 'react-redux';
import { useParams } from 'react-router-dom';

import { ButtonSend } from './components/ButtonSend';
import { ButtonDel } from './components/ButtonDel';
import { Message } from './components/Message';

import { addMessage } from 'src/store/chatlist/slice';

import style from './Form.module.scss';

export interface FormProps {
  userName: string;
}

export const Form = memo<FormProps>(({ userName }) => {
  const [value, setValue] = useState('');
  const dispatch = useDispatch();
  const { chatId } = useParams();

  const handleClickSubmit = (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();

    if (chatId && value) {
      dispatch(
        addMessage({ chatId, message: { text: value, author: userName } }),
      );
    }
    setValue('');
  };
  return (
    <form
      data-testid="form"
      onSubmit={handleClickSubmit}
      className={style.form}
    >
      <ButtonDel />
      <Message value={value} setValue={setValue} />
      <ButtonSend disabled={!userName} />
    </form>
  );
});
