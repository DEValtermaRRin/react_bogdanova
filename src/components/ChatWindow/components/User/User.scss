.user {
  display: flex;
  justify-content: space-between;
  align-items: center;
  width: 30px;
  height: 30px;
  background-color: #284779;
  overflow: hidden;
  border-radius: 15px;
  margin: 3px;
  &:hover {
    animation: user-dragout 0.5s ease-in-out 0.1s 1 normal forwards;
  }

  &:hover &__input {
    animation: input-dragout 0.5s ease-in-out 0.1s 1 normal forwards;
  }

  &__icon {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 30px;
    height: 30px;
    background: url(./assets/user.svg) 100%;
    background-color: #284779;
    border-radius: 50%;
    overflow: hidden;
    &:hover {
      cursor: pointer;
    }
  }
  &__input {
    height: 26px;
    border-radius: 15px;
    border: none;
    background: #252424;
    font-size: 14px;
    font-weight: 600;
    letter-spacing: 0.1rem;
    color: #366dc5;
    opacity: 0;
    width: 0;
    margin-right: 0px;
    &::placeholder {
      color: #4979af75;
    }
  }
}

@keyframes user-dragout {
  0% {
    width: 35px;
  }
  100% {
    width: 250px;
  }
}

@keyframes input-dragout {
  0% {
    padding: 0;
    opacity: 0;
    width: 0;
    margin-right: 0px;
  }
  100% {
    padding: 3px;
    opacity: 1;
    width: 200px;
    margin-right: 2px;
  }
}
