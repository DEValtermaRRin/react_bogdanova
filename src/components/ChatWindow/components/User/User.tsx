import React, { FC } from 'react';
import style from './User.module.scss';

interface UserProps {
  name: string;
  getName: (e: string) => void;
}

export const User: FC<UserProps> = ({ name, getName }) => (
  <div className={style.user}>
    <div className={style.user__icon}></div>
    <input
      className={style.user__icon}
      data-testid="user"
      value={name}
      onChange={(e) => getName(e.target.value)}
      type="text"
      placeholder="Enter your name"
    />
  </div>

  
);
