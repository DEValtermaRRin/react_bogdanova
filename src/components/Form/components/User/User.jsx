import React from "react";

export const User = ({ name, getName }) => (
  <input
    value={name}
    onChange={(e) => getName(e.target.value)}
    type="text"
    placeholder="Your name? "
  />
);
