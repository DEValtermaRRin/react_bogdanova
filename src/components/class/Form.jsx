import React, { Component } from 'react';
import { Input } from './Input';
import { Button } from './Button';
import { Message } from './Message';
import { ButtonClear } from './ButtonClear';

export class Form extends Component {
  state = {
    name: 'Send message',
    value: '',
    messages: [],
    btnClear: 'Clear messages',
  };

  handleClick = () => {
    this.setState({ messages: [...this.state.messages, this.state.value] });
    this.setState({ value: '' });
  };

  handleChange = (e) => {
    this.setState({ value: e.target.value });
  };

  clearMsg = () => {
    this.setState({ messages: [] });
  };

  render() {
    return (
      <>
        <Input change={this.handleChange} value={this.state.value} />
        <Button name={this.state.name} click={this.handleClick} />
        <Message send={this.state.messages} />
        <ButtonClear name={this.state.btnClear} click={this.clearMsg} />
      </>
    );
  }
}
