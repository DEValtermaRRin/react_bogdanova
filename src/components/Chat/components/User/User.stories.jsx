import React from 'react';
import { User } from './User';

export default {
  component: User,
};

const Template = (args) => <User {...args} />;

export const Test = Template.bind({});
