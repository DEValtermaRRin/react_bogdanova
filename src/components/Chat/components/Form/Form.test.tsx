import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Form } from './Form';
import { User } from '../User/User';

describe('Form', () => {
  it('render component', () => {
    const mockFn = jest.fn();
    render(
      <Form
        addMessages={Object}
        onSubmit={mockFn}
        userName={''}
        delMessages={mockFn}
      />,
    );
  });
  it('render with snapshot', () => {
    const mockFn = jest.fn();
    expect(
      render(
        <Form
          addMessages={Object}
          onSubmit={mockFn}
          userName={''}
          delMessages={mockFn}
        />,
      ),
    ).toMatchSnapshot();
  });
  it('render message with text', () => {
    const mockFn = jest.fn();
    render(
      <Form
        addMessages={Object}
        onSubmit={mockFn}
        userName={''}
        delMessages={mockFn}
      />,
    );
    fireEvent.input(screen.getByTestId('message'), {
      target: { value: 'Hello' },
    });
    fireEvent.click(screen.getByTestId('button'));
    expect(screen.getByText(/hello/i)).toBeInTheDocument();
  });
  it('check click', () => {
    const handleSubmit = jest.fn();
    render(<User name="Vasia" getName={handleSubmit} />);
    const { getByTestId } = render(
      <Form
        userName={''}
        addMessages={Object}
        onSubmit={handleSubmit}
        delMessages={handleSubmit}
      />,
    );
    const button = getByTestId('button');
    fireEvent.input(screen.getByTestId('user'), {
      target: { value: 'Vasya' },
    });
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });
});
