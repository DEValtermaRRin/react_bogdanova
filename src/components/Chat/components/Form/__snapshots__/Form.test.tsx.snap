// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form render with snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
        data-testid="form"
      >
        <button
          class="del-button"
          type="button"
        >
          <svg
            viewBox="0 0 96 96"
            xmlns="http://www.w3.org/2000/svg"
            xmlns:xlink="http://www.w3.org/1999/xlink"
          >
            <path
              d="m24,78c0,4.968 4.029,9 9,9h30c4.968,0 9-4.032 9-9l6-48h-60l6,48zm33-39h6v39h-6v-39zm-12,0h6v39h-6v-39zm-12,0h6v39h-6v-39zm43.5-21h-19.5c0,0-1.344-6-3-6h-12c-1.659,0-3,6-3,6h-19.5c-2.487,0-4.5,2.013-4.5,4.5s0,4.5 0,4.5h66c0,0 0-2.013 0-4.5s-2.016-4.5-4.5-4.5z"
            />
          </svg>
        </button>
        <textarea
          class="message"
          data-testid="message"
          placeholder="Your message"
        />
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled btn btn-top css-11qr2p8-MuiButtonBase-root-MuiButton-root"
          data-testid="button"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          Send message
          <span
            class="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SendIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
              />
            </svg>
          </span>
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
      data-testid="form"
    >
      <button
        class="del-button"
        type="button"
      >
        <svg
          viewBox="0 0 96 96"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
        >
          <path
            d="m24,78c0,4.968 4.029,9 9,9h30c4.968,0 9-4.032 9-9l6-48h-60l6,48zm33-39h6v39h-6v-39zm-12,0h6v39h-6v-39zm-12,0h6v39h-6v-39zm43.5-21h-19.5c0,0-1.344-6-3-6h-12c-1.659,0-3,6-3,6h-19.5c-2.487,0-4.5,2.013-4.5,4.5s0,4.5 0,4.5h66c0,0 0-2.013 0-4.5s-2.016-4.5-4.5-4.5z"
          />
        </svg>
      </button>
      <textarea
        class="message"
        data-testid="message"
        placeholder="Your message"
      />
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root Mui-disabled btn btn-top css-11qr2p8-MuiButtonBase-root-MuiButton-root"
        data-testid="button"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        Send message
        <span
          class="MuiButton-endIcon MuiButton-iconSizeSmall css-155nyw6-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SendIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.01 21 23 12 2.01 3 2 10l15 2-15 2z"
            />
          </svg>
        </span>
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
